File: 2021/01/part2.py
def part2(filename):
	with open(filename, 'r') as file:
		data = [int(l) for l in file.read().splitlines()]
	return sum([1 if data[i+2] > data[i-1] else 0 for i in range(1, len(data) - 2)])

if __name__ == '__main__':
	count = part2('input.txt')
	print(count)

File: 2022/03/part2.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func part2() {
	file, _ := os.Open("input.txt")
	scanner := bufio.NewScanner(file)
	defer file.Close()

	group := make([]string, 3)
	count := 0
	sum := 0
	for scanner.Scan() {
		group[count] = scanner.Text()
		count = (count + 1) % 3

		if count == 0 {
			ch := getCommonLetter(group...)
			sum += getPriority(ch)
		}
	}

	fmt.Println(sum)
}
